0chain/system_test
# Update Smart Contract Config Tests

Tests only testing an update of a single config for each smart contract.
Also fixes some other flaky tests
The tests are skipped if `sc_owner_wallet.json` does not exist on `./tests/cli_tests/config`. The wallet file is created on CI and pulled from GH secret `SMART_CONTRACT_OWNER_WALLET_JSON`

Test cases:

`fc-update-config`
- should allow update of max_pour_amount
- update max_pour_amount to invalid value should fail
- update by non-smartcontract owner should fail
- update with bad config key should fail
- update with missing keys param should fail
- update with missing values param should fail

`ip-update-config`
- should allow update of min_lock
- update min_lock to invalid value should fail
- update by non-smartcontract owner should fail
- update with bad config key should fail
- update with missing keys param should fail
- update with missing values param should fail

`mn-update-config`
- should allow update of interest_rate
- update interest_rate to invalid value should fail
- update by non-smartcontract owner should fail
- update with bad config key should fail
- update with missing keys param should fail
- update with missing values param should fail

`sc-update-config`
- should allow update of max_read_price
- update max_read_price to invalid value should fail
- update by non-smartcontract owner should fail
- update with bad config key should fail
- update with missing keys param should fail
- update with missing values param should fail

`vp-update-config`
- should allow update of max_destinations
- update max_destinations to invalid value should fail
- update by non-smartcontract owner should fail
- update with bad config key should fail
- update with missing keys param should fail
- update with missing values param should fail
